test/solvers/advanced/chambolle_pock_test.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""(Quasi-)Newton schemes to find zeros of functions (gradients)."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+# External
+import numpy as np
+import pytest
+
+# Internal
+import odl
+from odl.solvers import chambolle_pock_solver, f_cc_prox_l2_tv, g_prox_none
+from odl.util.testutils import all_almost_equal
+
+
+def test_proximal_factories():
+    """Test factory function creating the proximal operator."""
```

Comment:
Perhaps a line of theory here would be nice, Its not obvious where the examples come from and what we want to achieve.